src/AWSCredentials.jl

```diff
@@ -251,30 +251,72 @@ end
 
 using IniFile
 
-dot_aws_credentials_file() = get(ENV, "AWS_CONFIG_FILE",
+dot_aws_credentials_file() = get(ENV, "AWS_SHARED_CREDENTIALS_FILE",
                                  joinpath(homedir(), ".aws", "credentials"))
 
+dot_aws_config_file() = get(ENV, "AWS_CONFIG_FILE",
+                                 joinpath(homedir(), ".aws", "config"))
+
 """
-Load Credentials from [AWS CLI ~/.aws/credentials file]
+Load Credentials from [AWS CLI ~/.aws/credentials file] or [AWS CLI ~/.aws/config file]
 (http://docs.aws.amazon.com/cli/latest/userguide/cli-config-files.html).
 """
-
 function dot_aws_credentials()
 
-    @assert isfile(dot_aws_credentials_file())
-
-    ini = read(Inifile(), dot_aws_credentials_file())
+    @assert isfile(dot_aws_credentials_file()) || isfile(dot_aws_config_file())
 
     profile = get(ENV, "AWS_DEFAULT_PROFILE",
               get(ENV, "AWS_PROFILE", "default"))
+    
+    region = get(ENV, "AWS_DEFAULT_REGION", "us-east-1")
```

Comment:
Shouldn't be assuming a default region. If the profile doesn't have a region set or AWS_DEFAULT_REGION isn't set there should be an exception

Reply:
Should that also be the case in `aws_config`?

Reply:
I believe so. Maybe @samoconnor disagrees?

Reply:
I do :)
I have tried to include support for the env vars and config files that are uses by boto3 (and the AWS cli) as a convenience to end users. However, I'm not attempting to follow boto3's behaviour in every respect.

My intention is to try to make the Julia AWS interface as easy as possible for first time users. As such, I see having a default region as a feature that's good for people who don't know or care what a region is but just wan't to fetch some files from this thing called S3 where their collaborators keep their data.

Some examples:

- The S3 interface automatically [follows (and caches) bucket-region redirects](https://github.com/samoconnor/AWSS3.jl/blob/master/src/AWSS3.jl#L105) so that bucket regions can be completely ignored.

 - The list objects interface [handles pagination internally](https://github.com/samoconnor/AWSS3.jl/blob/master/src/AWSS3.jl#L503) rather than [requiring the user to deal with paginators](http://boto3.readthedocs.io/en/latest/guide/paginators.html) as boto3 does.